functions/v2/http_logging/main_test.py

```diff
@@ -45,7 +45,14 @@ def test_functions_log_http_should_print_message(app, capsys):
     # Force trace with trace header
     with app.test_request_context(headers={'x-cloud-trace-context': f"{mock_trace}/{mock_span};o=1"}):
         main.structured_logging(flask.request)
-        _, err = capsys.readouterr()
-        output_json = json.loads(err)
+        out, err = capsys.readouterr()
+
+        # In some situations, the library adds information before the first
+        # intended line. Use the next line in that case.
+        err_lines = err.splitlines()
+        if len(err_lines) > 1:
+            err = err_lines[1]
```

Comment:
Is this guaranteed to be at index 1? It might be safer to look at the list's content to find the right entry. Maybe using something like this:

```
err = next(log for log in err_lines if "Hello, world" in log)
```

Reply:
It seems to be guaranteed to be either the only line, or second if the instrumentation inserted a line before the actual output. I didn't want to go too far in searching for it.